test/lib/mockerode.ts

```diff
@@ -35,6 +35,26 @@ export type PartialImageInspectInfo = DeepPartial<
 	Id: string;
 };
 
+type Methods<T> = {
+	[K in keyof T]: T[K] extends (...args: any) => any ? T[K] : never;
+};
+
+function createFake<Prototype extends object>(prototype: Prototype) {
+	return (Object.getOwnPropertyNames(prototype) as Array<keyof Prototype>)
+		.filter((fn) => fn === 'constructor' || typeof prototype[fn] === 'function')
+		.reduce(
+			(res, fn) => ({
+				...res,
+				[fn]: () => {
+					throw Error(
+						`Fake method not implemented: ${prototype.constructor.name}.${fn}()`,
+					);
+				},
+			}),
+			{} as Methods<Prototype>,
+		);
+}
```

Comment:
Good idea! I checked real quick and the test fails with thrown error when a not-yet-implemented method is called in a suite.

Reply:
Never mind, I just saw your comment below in this commit (7db7ce74f1e429ede495ab5d72736c4af04339df) about avoiding silent fails